src/gurt/telemetry.c

```diff
@@ -2513,6 +2554,224 @@ get_unique_shmem_key(const char *path, int id)
 	return (key_t)d_hash_string_u32(salted, sizeof(salted));
 }
 
+static int
+shm_stat_key(key_t key, struct shmid_ds *shminfo, int *shmid_ptr)
+{
+	int shmid;
+	int rc;
+
+	if (unlikely(shminfo == NULL)) {
+		D_ERROR("NULL shminfo\n");
+		return -DER_INVAL;
+	}
+
+	rc = shmget(key, 0, 0);
+	if (rc < 0) {
+		D_ERROR("failed to get shmid for key 0x%x\n", key);
+		return rc;
+	}
+	shmid = rc;
+
+	rc = shmctl(shmid, IPC_STAT, shminfo);
+	if (rc != 0) {
+		D_ERROR("shmctl(%d, IPC_STAT) failed: %d\n", shmid, rc);
+		return rc;
+	}
```

Comment:
Yeah. In this case, I was thinking that the caller could deal with it, but you're right, better to return something appropriate here in case someone else uses it.

Reply:
Fixed.